sockeye/attention.py

```diff
@@ -310,6 +310,9 @@ def attend(att_input: AttentionInput, att_state: AttentionState) -> AttentionSta
                                               num_hidden=self.num_hidden,
                                               no_bias=True, name="%squery_hidden_fc" % self.prefix)
 
+            # scale down dot product by sqrt(num_hidden) [Vaswani et al, 17]
+            query *= self.num_hidden ** -0.5
```

Comment:
I wouldn't make this change backwards-incompatible, but rather add another parameter, or represent this as a different attention type like 'dot_scaled' (which could internally map to a parameter passed to DotAttention). Otherwise models that have been trained without the change will just silently suddenly compute a different attention score.